src/azul/service/manifest_controller.py

```diff
@@ -137,28 +140,37 @@ def get_manifest_async(self,
 
         if manifest is None:
             location = furl(self_url, args={'token': token.encode()})
+            command_lines = self.service.command_lines(manifest, str(location))
             body = {
                 'Status': 301,
                 'Location': location.url,
                 'Retry-After': token.wait_time
             }
         else:
-            if fetch:
+            try:
+                query_params['objectKey']
+            except KeyError:
                 url = self.manifest_url_func(fetch=False,
                                              catalog=manifest.catalog,
                                              format_=manifest.format_,
                                              filters=json.dumps(manifest.filters),
                                              objectKey=manifest.object_key)
             else:
                 url = manifest.location
+            command_lines = self.service.command_lines(manifest, url)
             body = {
                 'Status': 302,
                 'Location': url,
-                'CommandLine': self.service.command_lines(manifest, url)
+                'CommandLine': command_lines
             }
-
         if fetch:
             return Response(body=body)
         else:
             headers = {k: str(body[k]) for k in body.keys() & {'Location', 'Retry-After'}}
-            return Response(body='', status_code=body['Status'], headers=headers)
+            return Response(body=''.join(
+                dedent(f'''
+                    Download the manifest in {shell} with `curl` using:
+                    {cmd}
+                ''') for shell, cmd in sorted(command_lines.items())),
```

Comment:
Dictionaries now have a deterministic order. There is no need to sort them for reasons of determinism. If the output should be sorted, then the keys in the underlying dictionary should be ordered accordingly.

cc: @noah-aviel-dove